File: Cookbook_v2.Domain/RecipeModel/Recipe.cs
using System.Collections.Generic;
using Cookbook_v2.Toolkit.Domain.Abstractions;

namespace Cookbook_v2.Domain.RecipeModel
{
    public class Recipe : Entity
    {
        public static readonly int s_titleMaxLength = 50;
        public static readonly int s_descriptionMaxLength = 1000;
        public int UserId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public int TimesLiked { get; set; }
        public int TimesFavorited { get; set; }
        public int CookingTimeInMinutes { get; set; }
        public int ServingsCount { get; set; }
        public string ImageUrl { get; set; }
        public string Tags { get; set; }
        public virtual List<RecipeStep> RecipeSteps { get; set; }
        public virtual List<RecipeIngredientsSection> IngredientsSections { get; set; }
    }
}

File: Cookbook_v2.Site/ClientApp/src/app/interfaces/recipe.ts
import { IngredientsSection } from './ingredients-section';
import { RecipeStep } from './recipe-step';

export interface RecipeBase {
  userID: number;
  user: string;
  name: string;
  description: string;
  cookingTimeInMinutes: number;
  servingsAmount: number;
  tags: string[];
}

export interface RecipeWithImageFavsAndLikes {
  timesLiked: number;
  timesFavorited: number;
  imageURL: string;
}

export interface RecipeWithDetailedInfo {
  recipeSteps: RecipeStep[];
  ingredientsSections: IngredientsSection[];
}

export interface RecipePreview extends RecipeBase, RecipeWithImageFavsAndLikes {
  recipeID: number;
}

export interface Recipe extends RecipePreview, RecipeWithDetailedInfo {}
